[UI] Fix volume sorting/ auto-select

alexis-ld opened the pull request:
**Component**:
ui,volumes
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
There were some sorting issues on the Volumes list sorting (see #2920 )

- The auto-selected volume was not always the first one after health sorting
- The default sort sometime appeared in the URL

**Summary**:
Added a flag property to only auto-select when PVCs and Alerts are retrieved.
Removed URL sort when default sorted.

**Acceptance criteria**: 

The volumes list components waits for alerts data to be retrieved before auto-selecting the first volume.
The default sort does not appear in the URL anymore.


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2920 

<!-- If you want to refer to an issue while not closing it, use:


-->

Commit:
ui/volumes:
Wait until alerts and PVClist are retrieved before auto-selecting first volume based on health.

Refs: #2920

Commit:
ui/volumes:
Remove default sorting from query string on volumes list

Refs: #2920

bert-e commented:
# Hello alexis-ld,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2920-volume-sorting` into
`development/2.6`
* [w/2.7/bugfix/2920-volume-sorting](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/bugfix/2920-volume-sorting) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

ChengYanJin commented on ui/src/containers/VolumePage.js:
```diff
@@ -92,7 +92,7 @@ const VolumePage = (props) => {
   // If data has been retrieved and no volume is selected yet we select the first one
   useEffect(() => {
-    if (volumeListData.length && !currentVolumeName) {
+    if (volumeListData[0]?.volumeAlertsRetrieved && !currentVolumeName) {
```
```suggestion
    if (volumeListData.length && alerts.list.length && pVCList.length && !currentVolumeName) {
```

The `volumeAlertsRetrieved` is the flag to indicate if `PVC` and `alerts` are ready, add it for each item seem a bit redundant for me.

I would retrieve pVCList from VolumePage.  
```javascript
const pVCList = useSelector((state) => state?.app?.volumes?.pVCList);
```

You may want to add a little comment to quickly explain why we add this check/ what problem does it solve? 

Thanks.

alexis-ld replied:
Indeed :) thanks.

ChengYanJin commented on ui/src/services/NodeVolumesUtils.js:
```diff
@@ -244,6 +244,7 @@ export const getVolumeListData = createSelector(
       return {
         name: volume?.metadata?.name,
         node: volume?.spec?.nodeName,
+        volumeAlertsRetrieved :  alerts?.list && pVCList.length,
```
We can remove it ;)

ChengYanJin reviewed (changes requested):
Hello @alexis-ld ,
I see an improvement to do ;)

Thanks!

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

Commit:
ui/volumes:

Use a flag on the whole volumes instead of on each item to determine when alerts have been mapped

Refs: #2920

bert-e commented:
# History mismatch



Merge commit #90156c56d053c38e76401e3a89d0dd93d0d60691 on the integration branch
`w/2.7/bugfix/2920-volume-sorting` is merging a branch which is neither the current
branch `bugfix/2920-volume-sorting` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `bugfix/2920-volume-sorting` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2920-volume-sorting` into
`development/2.6`
* [w/2.7/bugfix/2920-volume-sorting](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/bugfix/2920-volume-sorting) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

bert-e commented:
# History mismatch



Merge commit #b8ccc9846a810b49c3ea34a932dabdd96830a15e on the integration branch
`w/2.7/bugfix/2920-volume-sorting` is merging a branch which is neither the current
branch `bugfix/2920-volume-sorting` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `bugfix/2920-volume-sorting` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexis-ld commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/2920-volume-sorting` into
`development/2.6`
* [w/2.7/bugfix/2920-volume-sorting](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/bugfix/2920-volume-sorting) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

ChengYanJin reviewed (approved):
LGTM! 
Thanks!

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

alexis-ld commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/84bd4b51514cad7bfc6c8eae341b78ddd81b2459) did not succeed in branch w/2.7/bugfix/2920-volume-sorting.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye alexis-ld.
